inst/include/cpp11/r_vector.hpp

```diff
@@ -793,40 +827,40 @@ inline typename r_vector<T>::iterator r_vector<T>::find(const r_string& name) co
 
 template <typename T>
 inline r_vector<T>::r_vector(const r_vector<T>& rhs)
-    : cpp11::r_vector<T>(safe[Rf_shallow_duplicate](rhs)),
-      protect_(detail::store::insert(data_)),
-      capacity_(rhs.capacity_) {}
+    : cpp11::r_vector<T>(safe[Rf_shallow_duplicate](rhs)), capacity_(rhs.capacity_) {}
 
 template <typename T>
 inline r_vector<T>::r_vector(r_vector<T>&& rhs)
-    : cpp11::r_vector<T>(rhs), protect_(rhs.protect_), capacity_(rhs.capacity_) {
-  rhs.data_ = R_NilValue;
-  rhs.protect_ = R_NilValue;
+    : cpp11::r_vector<T>(std::move(rhs)), capacity_(rhs.capacity_) {
+  rhs.capacity_ = 0;
```

Comment:
This should still have the same optimization through the `std::move()`, which should force the move constructor of read only `r_vector` to get called (I'll double check this)